File: fresh/spinner.py
import halo


class Spinner:
    def __init__(self):
        self.spinner = halo.Halo(spinner="dots")

    def start(self, text):
        self.spinner.start(text)

    def info(self, text):
        self.spinner.info(text)

    def stop(self):
        self.spinner.stop()

    def succeed(self):
        self.spinner.succeed()

    def fail(self):
        self.spinner.fail()

File: src/spinner.js
import ora from "ora";

export function createSpinner() {
  const spinner = ora({ spinner: "dots" });
  return {
    start(text) {
      spinner.start(text);
    },
    info(text) {
      spinner.info(text);
    },
    stop() {
      spinner.stop();
    },
    succeed(text) {
      spinner.succeed(text);
    },
    fail(text) {
      spinner.fail(text);
    },
  };
}
